type-metadata/src/lib.rs

```diff
@@ -1,7 +1,319 @@
-#[cfg(test)]
-mod tests {
-	#[test]
-	fn it_works() {
-		assert_eq!(2 + 2, 4);
+// Copyright 2019 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod registry;
+pub use registry::Registry;
+
+mod tests;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
+pub struct TypeIdent {
+	pub namespace: Vec<String>,
```

Comment:
I would make it `Vec<&'static str>` since we operate at compile-time anyway.

Reply:
Also I would encapsulate this into a custom type, like:
```rust
pub struct Namespace {
    namespace: Vec<&'static str>,
}

// some impl blocks for useful From and TryFrom impls
```

Reply:
Yep `&'static str` is more reasonable, will update. For the new `Namespace` type, how about `pub struct Namespace(Vec<&'static str>)`? As the field name `namespace` doesn't help much imho.

Reply:
I personally dislike tuple structs and only use them if they semantically and expressively encapsulate exactly one thing, e.g. `struct NonNullF32(f32);` or something like that.
However, this is a subjective topic and if you like `pub struct Namespace(Vec<'&static str>)` more than its named variant, then go for it.

Reply:
`segments` does make more sense, will update.